### PR TITLE
Feature/oe 470 create tests to upload file

### PR DESCRIPTION
**Pull request by ron-junior-devsquad**
## Review

For the submitter, initial self-review:

- [ ] Reviewed the test cases added for completeness and possible points for discussion
- [ ] Check the scope of the changeset - is everything in the diff required for the pull request?

For the pull request reviewer(s), this changeset has been reviewed for:

- [ ] Consistency between the changeset and the goal stated above
- [ ] Usage friction - is the proposed change in usage cumbersome or complicated?
- [ ] Complexity - is the proposed change introducing any unjustified complexity? 
- [ ] Thoroughness of added tests and any missing edge cases

### CODE REVIEW

File: tests/Feature/FileUpload/FileSectionUploadTest.php [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+use App\Models\TrainingPageSection;
+use App\Models\User;
+use Illuminate\Foundation\Testing\RefreshDatabase;
+use Illuminate\Http\UploadedFile;
+use Illuminate\Queue\Jobs\DatabaseJob;
+use Illuminate\Support\Collection;
+use Illuminate\Support\Facades\Storage;
+use Tests\TestCase;
+
+class FileUploadTest extends TestCase
```
Comment (devsquad-mathews-lima) [inline tests/Feature/FileUpload/FileSectionUploadTest.php:16]: The class name doesn't match the name of the file.

File: tests/Feature/FileUpload/FileSectionUploadTest.php [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        Storage::fake('local');
+
+        $this->files->push(UploadedFile::fake()->create('avatar.pdf'));
+
+        $this->post(route('uploadSectionFile', $this->section->id), [
+            'files' => $this->files,
+        ]);
+
+        Storage::disk('local')->assertExists('files/' . $this->department->id . '/' . $this->files->first()->hashName());
+    }
```
Comment (devsquad-mathews-lima) [inline tests/Feature/FileUpload/FileSectionUploadTest.php:57]: This test is a good scenario of **false assertive**. To crease a SectionFile it is necessary to provide **training_type**, and in the controller, this property is accessed by `request()->meta['training_type']`, and you're not passing this value in the body.

If you put the method `$this->withoutExceptionHandling();` you'll see that the test will fail since the request doesn't have this key.

**Review by devsquad-mathews-lima — CHANGES_REQUESTED** [review]
The test, in general, are okay, but I think you should test more scenarios like:

1. If you don't send an array of files, what should happen? The code will break?
2. Since the input in the `drop-file` component accept only the mime types `image/*,application/pdf`, what should happen if you upload an mp3/mp4/excel,etc?
3. What should happen if you don't provide the meta with the keyword `training_type`? Going further, what should occur if the provided value in the `training_type` isn't a valid value, something like `huehue`?

With that in mind, I think you should validate all the fields that are incoming.

**Review by devsquad-mathews-lima — APPROVED** [review]
